### PR TITLE
[cleanup] Hoist ModelTester objects to top level

### LINKED ISSUE #4902: [cleanup] Hoist ModelTester objects to toplevel
https://github.com/huggingface/transformers/pull/4046#issuecomment-628236744

many `ModelTester` objects are defined within classes. If we move them to the top level of the module, we can share code where possible and also have less complexity.

The task here is to move `ModelTester` objects to the top level.
Bonus: if the kwargs are never used, replace

```python
def__init(self, num_layers=2):
    self.num_layers=num_layers
```
with
```python
def__init(self):
    self.num_layers=2
```

### PR DESCRIPTION
**Pull request by aretius**
Fixes #4902 
@sshleifer 
All tests seem to pass. Please review!

### CODE REVIEW

File: tests/test_modeling_ctrl.py [hunk 1]
```diff
@@ -27,6 +27,140 @@
...
+            sequence_labels = ids_tensor([self.batch_size], self.type_sequence_label_size)
+            token_labels = ids_tensor([self.batch_size, self.seq_length], self.num_labels)
+            choice_labels = ids_tensor([self.batch_size], self.num_choices)
+
+        config = CTRLConfig(
+            vocab_size=self.vocab_size,
+            n_embd=self.hidden_size,
+            n_layer=self.num_hidden_layers,
+            n_head=self.num_attention_heads,
+            # intermediate_size=self.intermediate_size,
```
Comment (sshleifer) [inline tests/test_modeling_ctrl.py:86]: should these commented lines be deleted?

**Review by sshleifer — APPROVED** [review]
Wow, nice! 
@julien-c , you asked for this a few weeks back and @aretius made it happen!

**Comment by sshleifer** [comment]
I think we can investigate code reuse between `ModelTester`s in a separate PR, given the enormity of this one :)

**Comment by patrickvonplaten** [comment]
I like the changes! I only checked the files superficially. I trust @aretius and @sshleifer that no tests functionality was changed of got lost :-)

**Comment by aretius** [comment]
Thanks, @sshleifer :)
Would be glad to pick up code reuse in a separate PR 
Yes @patrickvonplaten no test functionality was lost

**Comment by aretius** [comment]
Merge conflict resolved @LysandreJik

**Review by patrickvonplaten — APPROVED** [review]

**Pull request merged**